### PR TITLE
terraform: support AWS marketplace images

### PR DESCRIPTION
**Pull request by msanft**
### Context

Along with Azure and GCP, we got approved for the AWS marketplace, which enables us to distribute Constellation's OS images through [said marketplace](https://aws.amazon.com/marketplace/pp/prodview-2mbn65nv57oys#pdp-usage).

### Proposed change(s)

- Add the field `.provider.aws.useMarketplaceImage` to `constellation-conf.yaml`. If set, this field builds the correct [AMI Alias](https://docs.aws.amazon.com/marketplace/latest/buyerguide/buyer-ami-aliases.html) from the selected image version. This AMI alias then can be used as a normal AMI, but will always resolve accordingly with respect to the region it's deployed to.
- Enable using AWS marketplace images through the Terraform provider.
- Document the necessary steps to use AWS marketplace images.

### Additional info

- [AB#3844](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3844)
- [E2E Test (Marketplace image)](https://github.com/edgelesssys/constellation/actions/runs/7756941077)
- [E2E Test (Non-marketplace image)](https://github.com/edgelesssys/constellation/actions/runs/7756946781)

### Checklist

- [x] Run the E2E tests that are relevant to this PR's changes
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -129,14 +129,17 @@ type AWSConfig struct {
 	//   AWS data center zone name in defined region. See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones
 	Zone string `yaml:"zone" validate:"required,aws_zone"`
 	// description: |
-	//   Name of the IAM profile to use for the control-plane nodes.
+	//   Name of the IAM ref/main/stream/nightly/v2.14.0-pre.0.20231201093533-432c4294c53profile to use for the control-plane nodes.
```
Comment (m1ghtym0) [inline internal/config/config.go:132]:
```suggestion
	//   Name of the IAM profile to use for the control-plane nodes.
```

**Review by m1ghtym0 — APPROVED** [review]
Top! That was quick:-)
I'll set the Marketplace listing to public to check if the link changes.

File: docs/docs/getting-started/marketplaces.md [hunk 2]
```diff
@@ -1,9 +1,23 @@
 # Using Constellation via Cloud Marketplaces
 
-Constellation is available through the Marketplaces of Azure and GCP. This allows you to create self-managed Constellation clusters that are billed on a pay-per-use basis (hourly, per vCPU) with your CSP account. You can still get direct support by Edgeless Systems. For more information, please [contact us](https://www.edgeless.systems/enterprise-support/).
+Constellation is available through the Marketplaces of AWS, Azure and GCP. This allows you to create self-managed Constellation clusters that are billed on a pay-per-use basis (hourly, per vCPU) with your CSP account. You can still get direct support by Edgeless Systems. For more information, please [contact us](https://www.edgeless.systems/enterprise-support/).
```
Comment (thomasten) [inline docs/docs/getting-started/marketplaces.md:3]:
```suggestion
Constellation is available through the Marketplaces of AWS, Azure, and GCP. This allows you to create self-managed Constellation clusters that are billed on a pay-per-use basis (hourly, per vCPU) with your CSP account. You can still get direct support by Edgeless Systems. For more information, please [contact us](https://www.edgeless.systems/enterprise-support/).
```

Comment (thomasten) [inline docs/docs/getting-started/marketplaces.md:3]: docs/overview/license.md needs the same change

File: docs/docs/getting-started/marketplaces.md [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
 
 This document explains how to run Constellation with the dynamically billed cloud marketplace images.
 
+## AWS
```
Comment (thomasten) [inline docs/docs/getting-started/marketplaces.md:7]: can we use csp tabs instead of sections?

File: docs/docs/getting-started/marketplaces.md [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
-Constellation is available through the Marketplaces of Azure and GCP. This allows you to create self-managed Constellation clusters that are billed on a pay-per-use basis (hourly, per vCPU) with your CSP account. You can still get direct support by Edgeless Systems. For more information, please [contact us](https://www.edgeless.systems/enterprise-support/).
+Constellation is available through the Marketplaces of AWS, Azure, and GCP. This allows you to create self-managed Constellation clusters that are billed on a pay-per-use basis (hourly, per vCPU) with your CSP account. You can still get direct support by Edgeless Systems. For more information, please [contact us](https://www.edgeless.systems/enterprise-support/).
 
 This document explains how to run Constellation with the dynamically billed cloud marketplace images.
 
-## Azure
+<tabs groupId="csp">
+<tabItem value="aws" label="AWS">
+
+On AWS, to use Constellation's marketplace images, ensure that you are subscribed to the [marketplace offering](https://aws.amazon.com/marketplace/pp/prodview-2mbn65nv57oys) through the web portal.
```
Comment (thomasten) [inline docs/docs/getting-started/marketplaces.md:10]:
```suggestion
To use Constellation's marketplace images, ensure that you are subscribed to the [marketplace offering](https://aws.amazon.com/marketplace/pp/prodview-2mbn65nv57oys) through the web portal.
```

File: docs/docs/getting-started/marketplaces.md [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
+Then, enable the use of marketplace images in your Constellation `constellation-conf.yaml` [config file](../workflows/config.md):
+
+```bash
+yq eval -i ".provider.aws.useMarketplaceImage = true" constellation-conf.yaml
+```
+
+</tabItem>
+<tabItem value="azure" label="Azure">
 
 On Azure, Constellation has a private marketplace plan. Please [contact us](https://www.edgeless.systems/enterprise-support/) to gain access.
```
Comment (thomasten) [inline docs/docs/getting-started/marketplaces.md:21]:
```suggestion
Constellation has a private marketplace plan. Please [contact us](https://www.edgeless.systems/enterprise-support/) to gain access.
```

File: docs/docs/getting-started/marketplaces.md [hunk 6]
```diff
@@ -20,11 +32,8 @@ Then, enable the use of marketplace images in your Constellation `constellation-
...
 
-Ensure that the cluster uses an official release image version (i.e., `.image=vX.Y.Z` in the `constellation-conf.yaml` file).
-
-From there, you can proceed with the [cluster creation](../workflows/create.md) as usual.
-
-## GCP
+</tabItem>
+<tabItem value="gcp" label="GCP">
 
 On GCP, to use a marketplace image, ensure that the account is entitled to use marketplace images by Edgeless Systems by accepting the terms through the [web portal](https://console.cloud.google.com/marketplace/vm/config/edgeless-systems-public/constellation).
```
Comment (thomasten) [inline docs/docs/getting-started/marketplaces.md:38]:
```suggestion
To use a marketplace image, ensure that the account is entitled to use marketplace images by Edgeless Systems by accepting the terms through the [web portal](https://console.cloud.google.com/marketplace/vm/config/edgeless-systems-public/constellation).
```

**Review by thomasten — APPROVED** [review]

**Pull request merged**